runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +215,48 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageSignature(String packageName, String fingerprint) {
+        Signature[] signatures = null;
+        try {
+            PackageManager packageManager = XWalkApplication.getApplication().getPackageManager();
+            PackageInfo packageInfo = packageManager.getPackageInfo(
+                    packageName, PackageManager.GET_SIGNATURES);
+            signatures = packageInfo.signatures;
+        } catch (NameNotFoundException e) {
+            Log.d(TAG, packageName + " not found");
+            return false;
+        }
+
+        if (signatures == null) {
+            Log.d(TAG, "No signature in package");
+            return false;
+        }
+
+        for (int i = 0; i < signatures.length; ++i) {
+            Log.d(TAG, "Checking signature " + i);
+            byte[] signature = signatures[i].toByteArray();
+
+            try {
+                MessageDigest md = MessageDigest.getInstance("SHA1");
+                StringBuffer buffer = new StringBuffer();
+                for (byte code : md.digest(signature)) {
+                    buffer.append(String.format("%02X", code));
+                }
+                String result = buffer.toString();
+                if (!result.equals(fingerprint)) {
+                    Log.d(TAG, "Fingerprint not match, " + result);
+                    continue;
+                }
+            } catch (NoSuchAlgorithmException e) {
```

Comment:
Is it possible that Java has no support for SHA1 these days? If that's the case shouldn't you at least add a log message to inform the user why the check failed?